cypress/e2e/hub/execution-environments.cy.ts

```diff
@@ -1,14 +1,60 @@
 import { hubAPI } from '../../support/formatApiPathForHub';
 import { ExecutionEnvironments } from './constants';
+import { randomString } from '../../../framework/utils/random-string';
 
-describe.skip('Execution Environments', () => {
+describe('Execution Environments', () => {
+  let remoteRegistryName = '';
   before(() => {
     cy.hubLogin();
   });
 
-  it('it should render the execution environments page', () => {
+  beforeEach(() => {
+    remoteRegistryName = `remote_registry_${randomString(3, undefined, { isLowercase: true })}`;
+
+    cy.createRemoteRegistry(remoteRegistryName);
+  });
+
+  it('can render the execution environments page', () => {
+    cy.navigateTo('hub', ExecutionEnvironments.url);
+    cy.verifyPageTitle(ExecutionEnvironments.title);
+  });
+
+  it('can add and delete a new execution environment', () => {
+    const eeName = `execution_environment_${randomString(3, undefined, { isLowercase: true })}`;
+    const upstreamName = `upstream_name_${randomString(3, undefined, { isLowercase: true })}`;
     cy.navigateTo('hub', ExecutionEnvironments.url);
     cy.verifyPageTitle(ExecutionEnvironments.title);
+    cy.getByDataCy('add-execution-environment').click();
+    cy.getByDataCy('name').type(eeName);
+    cy.getByDataCy('upstream-name').type(upstreamName);
+    cy.contains('[data-ouia-component-id="menu-select"]', 'Select registry')
+      .click()
+      .then(() => {
+        cy.contains('button[type="button"]', `${remoteRegistryName}`).click();
+      });
+    cy.getByDataCy('Submit').click();
+    cy.url().should('contain', '/execution-environments/');
+    cy.hubListFilter(eeName);
+    cy.get('tbody').find('tr').should('have.length', 1);
+    cy.get('tbody').within(() => {
+      cy.getByDataCy('container-repository-name-column-cell').should('contain', eeName);
+      cy.get('[data-cy="actions-dropdown"]')
+        .click()
+        .then(() => {
+          cy.get(`[data-cy="delete-environment"]`).click();
+        });
+    });
+    cy.intercept('DELETE', hubAPI`/v3/plugin/execution-environments/repositories/${eeName}/`).as(
+      'deleted'
+    );
+    cy.get('[data-ouia-component-id="Permanently delete execution environments"]').within(() => {
+      cy.get('[data-ouia-component-id="confirm"]').click();
+      cy.get('[data-ouia-component-id="submit"]').click();
+      cy.clickButton('Close');
+    });
+    cy.wait('@deleted').then((deleted) => {
+      expect(deleted.response?.statusCode).to.eql(202);
```

Comment:
Verifying that a 202 was returned does not mean that the EE was deleted, but rather that a task was created. Ideally, the status of the action should be verified. For the user, the status of the HTTP code here does not matter. If the API changes and sends back a 204, as it does in other places, the feature will still behave the same way for the use